message_app_client/src/components/conversation.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+
+class ConversationCard extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            lastMessageContent: null,
+            conversationStatus: null,
+            title: null,
+            conversationID: null,
+        }
+    }
+
+    componentDidMount(){
+        this.setState({
+            lastMessageContent: this.props.lastMessageContent,
+            conversationStatus: this.props.conversationStatus,
+            title: this.props.title,
+            conversationID: this.props.conversationID
+        });
+    }
+
+    componentDidUpdate(prevProps){
+        // Update the state when the conversation has a new lastMessage
+        if (this.props.lastMessageContent !== prevProps.lastMessageContent){
+            this.setState({
+                lastMessageContent: this.props.lastMessageContent
```

Comment:
This is quite wrong, because two messages can have the same content. Instead, you can store the message ID. Then, we can compare the IDs of those messages, and will make 1 more request to fetch the content of the latest one to display on the conversation card. It is not really costly, I think.